test/test_campaign.js

```diff
@@ -0,0 +1,194 @@
+const _deploy_contracts = require("../migrations/2_deploy_contracts");
+const truffleAssert = require("truffle-assertions");
+const BigNumber = require('bignumber.js'); // npm install bignumber.js
+const oneEth = new BigNumber(1000000000000000000); // 1 eth
+
+var assert = require("assert");
+var IAM = artifacts.require("../contracts/IAM.sol");
+var Campaign = artifacts.require("../contracts/TestCampaign.sol");
+var CampaignFactory = artifacts.require("../contracts/TestCampaignFactory.sol");
+
+contract ("Campaign", function(accounts){
+    before( async() => {
+        IAMInstance = await IAM.deployed();
+        campaignFactoryInstance = await CampaignFactory.deployed();
+    });
+
+    let camp1;
```

Comment:
Perhaps this variable declaration can be shifted to just before line 65, where it is first used in test case `CAMFAC02` for better code readability

Reply:
Declaration shifted.